### ISSUE TITLE
CustomConstraint is not working ?

### ISSUE THREAD

**Issue by JulienGervai** [sdv-dev/SDV#235]
* SDV version: 0.4.5
* Python version: 3.8.3
* Operating System: Windows 10

### Description

I am trying to use CustomConstraint to limit the values that the different columns can take. I base my work on https://github.com/sdv-dev/SDV/issues/200 

### What I Did

I define the function is valid like that : 

![is_valid](https://user-images.githubusercontent.com/69855022/98095499-f9312600-1e8a-11eb-8338-a452454e83b6.PNG)

So the values in a columns can't be less than the minimum in the original datafame
Then I define the CustomConstraint like that : 

![customconstraint](https://user-images.githubusercontent.com/69855022/98095627-1c5bd580-1e8b-11eb-898f-fafe4e037198.PNG)

I train ctgan : 

![train](https://user-images.githubusercontent.com/69855022/98095737-3b5a6780-1e8b-11eb-9ab4-ccb7b36d9fd0.png)

Generate fake data : 

![sample](https://user-images.githubusercontent.com/69855022/98095816-54fbaf00-1e8b-11eb-97c9-bed2dd2ba8e4.PNG)
And I take a look at the results : 

![results](https://user-images.githubusercontent.com/69855022/98095947-79578b80-1e8b-11eb-9636-59ae1f9a328f.PNG)

But we can see that a lot of values do not respect the minimum value.

**Comment by JulienGervai** [comment]
Moreover, if I use the "is_valid" function again on the synthetic data, I got : 

![is_valid_again](https://user-images.githubusercontent.com/69855022/98096472-16b2bf80-1e8c-11eb-9faa-2fcf86d7f10a.PNG)

So the is_valid function tell that there are some False values but we generate that synthetic data anyway

**Comment by csala** [comment]
Hi @JulienGervai 

I suspect that the problem is that you are returning a `pd.Series` with a freshly created index which is not aligned with the original one.

I your last example, you can notice that if you compare the index of the `synth_train` table with the index of the output from the `is_valid` call. They probably have the same length but a different sequence of numbers, which prevents the constraint filtering from working properly. I'll now open a new issue to be more robust against different types of outputs from the `is_valid` function, but in the meantime you can simply change the `pd.Series` to a `np.asarray` and everything should work.

So, basically, change this line:

```python
pd.Series(valid)
```

to this:

```python
np.asarray(valid)
```

Please let me know if it works after this change!

**Comment by JulienGervai** [comment]
You are totally right, my problem just vanished with that solution.
Thank you so much @csala !

**Comment by fra-cap** [comment]
Hi, 
I have the same problem but changing to np.asarray didn't fix it. 
`
def is_valid_not_payed(table_data):
    return_valid = [ ( fact == 'Non facturee') == (date =='00/01/00')
                for fact,date in zip(table_data['Etat Facturation BC'],table_data['Date de solde']) ]
  
    return np.asarray(return_valid)

not_payed_constraint = CustomConstraint(
    columns = [['Etat Facturation BC','Date de solde']],
    is_valid = is_valid_not_payed
)` 

Do you have any idea of what the problem could be?

I tried to return a true value and it works but if a return False it raises sdv.errors.ConstraintsNotMetError
